examples/large_models/tp_llama/REAME.md

```diff
@@ -0,0 +1,132 @@
+
+```bash
+python ../utils/Download_model.py --model_name meta-llama/Llama-2-7b
+```
+The script prints the path where the model is downloaded as below.
+
+`model/models--meta-llama--Llama-2-7b/snapshots/365ffa8f1a6c455d3e2028ae658236b4b85ba824`
+
+
+### Step 3: Convert the "Meta" checkpoints to PyTorch Distributed compliant checkpoints
+
+Convert the checkpoints to  PT-D compliant checkpoints as follows, note that for 7B `--model_parallel_size 1` for 13B would be `--model_parallel_size 2` and 70B `model_parallel_size 8`, you can also set `--nproc_per_node ` accordingly. PT-D compliant support flexible world_size when loading back the checkpoints into TP(lized) model. 
+
+You would be able to use larger number of processes/ TP size when load the model back. For example if you have converted the `13B` checkpoints with `--nproc_per_node 2`, during the inference you can use `--nproc_per_node` be `[2, max_num_available_gpu]` which you are changing the world_size and effectively the TP size. The recommendation here is to keep the TP size as shown above respective to model size, 7B (TP Size =1), 13B (TP Size =2), 70B (TP Size =8), unless your benchmark and your batch size/ compute load compensate for communication cost.
+
+
+This will save the model args in `model_args.json`, during the inference step you need to pass this json file for build the model. Make sure you are setting  `--max_seq_len` which is the maximum sequence length for input text (context length) and `--max_batch_size` which is maximum batch size for inference to respective values. These two values will be used to construct the KV cache.
+
+```
+torchrun --nnodes 1 --nproc_per_node 8 convert_checkpoints.py --original_ckpt_dir  PATH/TO/MODEL/CHECKPOINTS  --tokenizer_path PATH/TO/MODEL/CHECKPOINTS/tokenizer.model --model_parallel_size 1 --save_checkpoint_dir converted_checkpoints --max_seq_len 512 --max_batch_size 2
+
+```
+
+
+
+### Step 4: set up the configs:
+
+Lets setup configs in `model-config.yaml` 
+
+```
+#frontend settings
+minWorkers: 1
+maxWorkers: 1
+maxBatchDelay: 200
+responseTimeout: 300
+parallelType: "tp"
+deviceType: "gpu"
+
+torchrun:
+    nproc-per-node: 8 # TP size
+
+handler:
+    converted_ckpt_dir: "PATH/TO/converted_checkpoints"
+    tokenizer_path: "/PATH/TO/MODEL/CHECKPOINTS/tokenizer.model"
+    model_args_path: "PATH/TO/model_args.json"
+    max_seq_len: 512
+    max_batch_size: 6
+    max_new_tokens: 50
+    temperature: 0.6
+    top_p: 0.9
+    manual_seed: 40
+    mode: "text_completion" #choices are text_completion, chat
+```
+
+### step 5: Create the mar file:
+Create the mar file using the following command here. 
+
+```
+ torch-model-archiver --model-name llama --version 1.0 --handler llama-handler.py --config-file model-config.yaml --archive-format tgz --extra-files "llama2.py,llama2_tokenizer.py,generate.py,checkpoint_converter.py" 
```

Comment:
could you change to:

torch-model-archiver --model-name llama --version 1.0 --handler llama-handler.py --config-file model-config.yaml --archive-format no-archive --extra-files "llama2.py,llama2_tokenizer.py,generate.py,checkpoint_converter.py"

mv TO llama/

Reply:
@lxning moving the code to llama mess up with dependencies the reason is we have two stages, stage one is checkpoint conversion which means we need make that folder as a module then next step is serving which mean we need to bundles these files to mar file when we extracted in /tmp then it would be very messy to access those files.

Reply:
don't cp. you can just exactly copy my comments